pkg/types/workload_test.go

```diff
@@ -0,0 +1,356 @@
+		ObjectMeta: slim_metav1.ObjectMeta{
+			Name:         "foo-xxddxx",
+			GenerateName: "foo-",
+			Namespace:    "foo",
+			Labels: map[string]string{
+				"foo":  "xxddxx",
+				"foo1": "xdx",
+			},
+			OwnerReferences: []slim_metav1.OwnerReference{
+				{
+					Kind:       "Deployment",
+					Name:       "foo",
+					Controller: &controller,
+				},
+			},
+		},
+	}
+	w, err := GetWorkloadDataFromPod(k8sPod)
+	if err != nil {
+		c.Fatal(err)
+	}
+	c.Assert(*w, checker.Equals, expectedWorkload)
+}
+
+func (s *WorkloadSuite) TestGetWorkloadFromModel(c *check.C) {
+	expectedWorkload := Workload{
+		Name:      "foo",
+		Namespace: "foo",
+		Kind:      "Deployment",
+	}
+
+	ciliumEndpoint := cilium_v2.CiliumEndpoint{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:         "foo-xxddxx",
+			GenerateName: "foo-",
+			Namespace:    "foo",
+			Labels: map[string]string{
+				"foo":  "xxddxx",
+				"foo1": "xdx",
+			},
+		},
+		Status: cilium_v2.EndpointStatus{Workload: &models.Workload{
+			Name:      "foo",
+			Namespace: "foo",
+			Kind:      "Deployment",
+		}},
+	}
+	w := GetWorkloadFromModel(ciliumEndpoint.Status.Workload)
+
+	c.Assert(*w, checker.Equals, expectedWorkload)
+}
+
+func TestDeploymentMetadata(t *testing.T) {
+	controller := true
+	tests := []struct {
+		name         string
+		pod          *slim_corev1.Pod
+		wantWorkload *Workload
+		expectOK     bool
+	}{
+		{
+			name: "deployment-name-deploy",
+			pod:  podForDeployment("deploy", "12345", "default"),
+			wantWorkload: &Workload{
+				Name:      "deploy",
+				Kind:      "Deployment",
+				Namespace: "default",
+			},
+			expectOK: true,
+		},
+		{
+			name: "deployment-name-deploy2",
+			pod:  podForDeployment("deploy2", "45678", "default"),
+			wantWorkload: &Workload{
+				Name:      "deploy2",
+				Kind:      "Deployment",
+				Namespace: "default",
+			},
+			expectOK: true,
+		},
+		{
+			name: "non-deployment",
+			pod: &slim_corev1.Pod{
+				ObjectMeta: slim_metav1.ObjectMeta{
+					Name:         "replicaset0-abcd0",
+					GenerateName: "replicaset0-",
+					Namespace:    "default",
+					OwnerReferences: []slim_metav1.OwnerReference{{
+						Controller: &controller,
+						Kind:       "ReplicaSet",
+						Name:       "replicaset0",
+					}},
+				},
+			},
+			wantWorkload: &Workload{
+				Name:      "replicaset0",
+				Kind:      "ReplicaSet",
+				Namespace: "default",
+			},
+			expectOK: true,
+		},
+		{
+			name: "bare-pod",
+			pod: &slim_corev1.Pod{
+				ObjectMeta: slim_metav1.ObjectMeta{
+					Name: "some-pod",
+				},
+			},
+			expectOK: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			workload, err := GetWorkloadDataFromPod(tt.pod)
+			if tt.expectOK != (err == nil) {
+				t.Fatalf("expected ok=%t, got ok=%t, return err shows %s", tt.expectOK, err == nil, err.Error())
+			}
+			if err == nil {
+				if !reflect.DeepEqual(workload, tt.wantWorkload) {
```

Comment:
This is a good spot to use your equality method. Which would be good to test as well.